bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/ghost/Ghost.java

```diff
@@ -0,0 +1,358 @@
+        addComponent(_cableThrougthput);
+        _fixedOptics = new TransmissionElement(getDirectory()+"/" + Ghost.INSTR_PREFIX + "fixedOptics"+getSuffix());
+        _fixedOptics.setDescription("Fix Optics");
+        addComponent(_fixedOptics);
+
+        _gratingOptics = new GhostGratingOptics(
+                getDirectory() + "/" + Ghost.INSTR_PREFIX,
+                gp.resolution().get_displayValue(),                        // ghost_{SR,HR}.dat
+                "gratings",                                                // ghost_gratings.dat
+                gp.centralWavelength().toNanometers(),
+                _detector.getDetectorPixels(),
+                gp.spectralBinning().getValue());
+        _gratingOptics.setDescription("Grating Resolution");
+        addDisperser(_gratingOptics);
+        _sampling = super.getSampling();
+        _ifu = new IFUComponent(gp.resolution());
+        addComponent(_ifu);
+
+        _ghostSaturLimitWarning = new GhostSaturLimitRule(AD_SATURATION, WellDepth, getSpatialBinning(), getSpectralBinning(), gain() , 0.90);
+
+        _ghostResolution = new TransmissionElement(getDirectory()+"/" + Ghost.INSTR_PREFIX + gp.resolution().get_displayValue()+ "_perResolution" +getSuffix());
+    }
+
+    public GhostType.DetectorManufacturer getDetManufacture() {
+        return _ccdColor;
+    }
+
+    /**
+     * Returns an array containing this instrument, or, if there are multiple detector CCDs,
+     * an array containing instances of this instrument with the CCD set differently
+     * (Used to implement hamamatsu CCD support).
+     */
+    public Ghost[] getDetectorCcdInstruments() {
+        return _instruments;
+    }
+
+    /**
+     * Index of current CCD in detector
+     *
+     * @return 0, 1, or 2 when there are multiple CCDs in the detector (default: 0)
+     */
+
+    /**
+     * Returns the name of the detector CCD
+     */
+    public String getDetectorCcdName() {
+        return _detector.getName();
+    }
+
+    /**
+     * Returns the effective observing wavelength.
+     * This is properly calculated as a flux-weighted averate of
+     * observed spectrum.  So this may be temporary.
+     *
+     * @return Effective wavelength in nm
+     */
+    public int getEffectiveWavelength() {
+        Log.fine("ghost getEffectiveWavelength: "+ _gratingOptics.getEffectiveWavelength());
+        return (int) _gratingOptics.getEffectiveWavelength();
+
+    }
+
+    public double getGratingDispersion() {
+        Log.fine("dispersion: " + _gratingOptics.dispersion(-1));
+        return _gratingOptics.dispersion(-1);
+    }
+
+    /**
+     * Returns the subdirectory where this instrument's data files are.
+     */
+    public String getDirectory() {
+        return ITCConstants.LIB + "/" + INSTR_DIR;
+    }
+
+    @Override
+    public double wellDepth() {
+        Log.info("TODO. VENU has to give this information. Not implemented yet Ghost.java wellDepth ");
```

Comment:
Above it was indicated that the value was confirmed, is it?

Reply:
No, this value never was confirmed it.